### PR TITLE
Introduce SYSTEMS, CONST and NODES as replacement for system_, const_…

### PR DESCRIPTION
**Pull request by j-p-a-u-l**
… and _node

Old variants are still available for backwards compatibility

Fixes Issue #26

### CODE REVIEW

**Comment by fkusei** [comment]
Can't we get rid of `hostname!` now?

**Comment by j-p-a-u-l** [comment]
> Can't we get rid of `hostname!` now?

How should this work? The only idea I have, would be to catch the Exception, if the node does not exist and assume a hostname in this case. But I am not sure, whether that's a good idea. It's a "better exclicit than implicit" thing.

As far as I know there is no restriction in choosing the node name in BW. This could lead to the case, that you cannot distinguish from reading the Automatix script file, whether a system is teated as Bundlewrap or not Bundlewrap system.

**Comment by fkusei** [comment]
> How should this work? The only idea I have, would be to catch the Exception, if the node does not exist and assume a hostname in this case.

Yeah, that's basically what I would do:
```python
def get_ssh_target_for_system(system):
    try:
        return NODE.system.hostname
    except NoSuchNode:
        return system
```

> This could lead to the case, that you cannot distinguish from reading the Automatix script file, whether a system is treated as Bundlewrap or not Bundlewrap system.

Yeah, that could be a problem. Personally i would prefer not having the `hostname!` prefix, because my nodes are named in a way one could never confuse them with a "normal" hostname. Can't speak for everyone though.

File: README.md [hunk 1]
```diff
@@ -178,7 +178,7 @@ The **scriptfile** has to contain valid YAML.
 ### EXAMPLE: scriptfile
     
     name: Migration Server XY
-    # Systems you like to refer to in pipeline (accessible via 'system_source')
+    # Systems you like to refer to in pipeline (accessible via 'SYSTEMS.source')
     # If Bundlewrap support is activated use node names instead of hostnames.
```
Comment (fkusei) [inline README.md:182]: 'you can also use node names' (because we now also support combinations of node names and hostnames)

**Comment by j-p-a-u-l** [comment]
> Yeah, that could be a problem. Personally i would prefer not having the `hostname!` prefix, because my nodes are named in a way one could never confuse them with a "normal" hostname. Can't speak for everyone though.

What about turning things around?

```yaml
systems:
    mynode: bw!mynode
    myhostname: myhostname
```

But this would break backwards compatibility.

File: automatix/bundlewrap.py [hunk 2]
```diff
@@ -27,3 +29,12 @@ def _get_remote_hostname(self):
...
         node = self.env.config['bw_repo'].get_node(system)
         return node.hostname
+
+
+class BWNodesWrapper:
+    def __init__(self, repo: Repository, systems: dict):
+        self.repo = repo
+        self.systems = systems
+
+    def __getattr__(self, name):
```
Comment (vanadinit) [inline automatix/bundlewrap.py:39]: Just recognized, that this seems to be a Python3.7 feature: We should update the requirements here!

Comment (j-p-a-u-l) [reply automatix/bundlewrap.py:39]: FTR: Python3.7 feature is for modules. So it also works this way in Python3.6. No need for requirement adjustment. I am going to revert the commit.

**Review by nkoch-ops — APPROVED** [review]

**Pull request merged**